packages/edit-navigation/src/store/actions.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * External dependencies
+ */
+import { invert, keyBy, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { select, resolveMenuItems, dispatch, apiFetch } from './controls';
+import { uuidv4, menuItemsQuery, KIND, POST_TYPE } from './utils';
+
+// Hits POST /wp/v2/menu-items once for every Link block that doesn't have an
+// associated menu item. (IDK what a good name for this is.)
+export const createMissingMenuItems = serializeProcessing( function* ( post ) {
+	const menuId = post.meta.menuId;
+	const mapping = post.meta.menuItemIdToClientId;
+	const clientIdToMenuId = invert( mapping );
+
+	const stack = [ post.blocks[ 0 ] ];
+	while ( stack.length ) {
+		const block = stack.pop();
+		if ( ! ( block.clientId in clientIdToMenuId ) ) {
+			const menuItem = yield apiFetch( {
+				path: `/__experimental/menu-items`,
+				method: 'POST',
+				data: {
+					title: 'Placeholder',
+					url: 'Placeholder',
+					menu_order: 0,
+				},
+			} );
+
+			mapping[ menuItem.id ] = block.clientId;
+			const menuItems = yield resolveMenuItems( menuId );
+			yield dispatch(
+				'core',
+				'receiveEntityRecords',
+				'root',
+				'menuItem',
+				[ ...menuItems, menuItem ],
```

Comment:
We want to:

1. Append another entry to the store
1. Append entry ID to stored query results

Using just `menuItem` does 1) but breaks 2) as instead of appending it replaces the first ID. Using `[ ...menuItems, menuItem ]` both appends the new entry to `queriedData.items` AND appends entry ID to stored query results to `queriedData.queries["menus=123"]`.